### PR TITLE
chore: Update Istanbul to NYC and bump Mocha

### PR DESCRIPTION
**Pull request by nschonni**
- Update mocha to latest, which drops the the `npm audit` warnings
- Removes the depreciated istanbul package with nyc
- Drops Coveralls, since we can just use it only in CI now
- Coverage does decrease, since it now globs in more of the "scripts", not currently covered by tests

### CODE REVIEW

File: .nycrc.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "all": true,
+  "include": [
+    "bin/*",
```
Comment (nschonni) [inline .nycrc.json:4]: Don't think I quite got this right, but I also don't think it was working before either

**Review by xzyfer — APPROVED** [review]

**Comment by nschonni** [comment]
Oops, should have waited for CI to finish. I think Windows isn't exited properly now. Quick search sounds like it's related to uncleaned up references so I'll see what I can find

**Comment by nschonni** [comment]
OK, narrowed it down to the CLI tests. I'll try troubleshooting it a bit before resorting to their hammer of `--exit` https://mochajs.org/#-exit

**Comment by nschonni** [comment]
OK, found an "easy" fix, of updating cross-spawn. PR incoming

**Pull request merged**